c-dependencies/js-compute-runtime/js-compute-builtins.cpp

```diff
@@ -997,9 +1047,30 @@ namespace RequestOrResponse {
     return true;
   }
 
-  bool body_source_pull_algorithm(JSContext* cx, CallArgs args, HandleObject stream,
-                                  HandleObject owner, HandleObject controller)
+  bool body_source_pull_algorithm(JSContext* cx, CallArgs args, HandleObject source,
+                                  HandleObject body_owner, HandleObject controller)
   {
+    // If the stream has been piped to a TransformStream whose readable end was then passed
```

Comment:
Is there any sort of reordering issue that could occur here, because of the multi-step discovery of the shortcut? What I mean is that the flow seems to be (i) pipe the stream to the body output, setting this property under the hood; (ii) later, the "pull algorithm" discovers this and does the direct-path optimization. But what if multiple streams are piped to the output, hitting step (i) for multiple sources, and then the pulls happen in some other arbitrary order?

Perhaps this doesn't happen either because the pull is performed right away, or the output has some mutual exclusion; I don't know enough about the overall APIs to say. But a comment somewhere describing how order of multiple appends is preserved would be helpful :-)

Reply:
Great question! That is indeed ruled out by how piping works: `pipeTo` locks the target stream, and it'll only unlock once the input stream closes, at which point you can call `pipeTo` with the next stream. Does that address your concern? If so, I'll add that to the comments somewhere.

Reply:
Ah, yes, that makes perfect sense. A note somewhere in the source making that connection would be great, thanks!